jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1565,8 +1565,8 @@ YUI.add('juju-view-utils', function(Y) {
     @param {String} region The cloud region to deploy to.
   */
   utils.deploy = function(
-    env, jem, users, autoPlaceUnits, createSocketURL, appSet, committed,
-    callback, autoplace=true, model, credential, cloud, region) {
+    env, controllerAPI, autoPlaceUnits, createSocketURL, appSet, committed,
```

Comment:
It seems to me that this function really receives too many arguments, and I believe that, when this function is called, understanding the arguments would be pretty hard. I'd like to refactor this so that it takes an object, with clearly defined and documented fields, and some documented default fallbacks in case the field is not specified. That would also be more future proof in cae we want to add another field.